src/audio/pipeline.c

```diff
@@ -690,6 +677,43 @@ static int pipeline_xrun_handle_trigger(struct pipeline *p, int cmd)
 	return ret;
 }
 
+static void pipeline_schedule_triggered(struct pipeline_walk_context *ctx,
+					int cmd)
+{
+	struct pipeline *p;
+	uint32_t flags;
+	uint32_t i;
+
+	irq_local_disable(flags);
+
+	for (i = 0; i < ctx->pipeline_count; ++i) {
+		p = ctx->pipelines[i];
+		if (!p)
+			continue;
+
+		switch (cmd) {
+		case COMP_TRIGGER_PAUSE:
+		case COMP_TRIGGER_STOP:
+		case COMP_TRIGGER_XRUN:
+			pipeline_schedule_cancel(p);
+			p->status = COMP_STATE_PAUSED;
```

Comment:
I assume p is only ever written to on this core? The outside irq_local_disable/enable is really to prevent being interrupted while walking the pipelines array?

Reply:
Yes, exactly.

Reply:
Just wanted to ensure all assumptions behind the code are correct.